src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    public static final int DEFAULT_CAPACITY = 16;
+    public static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] storage;
+    private int capacity;
+    private int size;
+    private int threshold;
+
+    private class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        private Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (capacity == 0) {
+            storage = new Node[DEFAULT_CAPACITY];
+            capacity = DEFAULT_CAPACITY;
+            threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+        }
```

Comment:
maybe you should do it in constructor?

Reply:
Maybe, but as I understand, we need to avoid implementing logic in constructors.

Reply:
do it in the constructor, this is not business logic but just a field initialization